src/hooks/use-site-details.tsx

```diff
@@ -159,11 +160,50 @@ export function SiteDetailsProvider( { children }: SiteDetailsProviderProps ) {
 
 	const createSite = useCallback(
 		async ( path: string, siteName?: string ) => {
-			const data = await getIpcApi().createSite( path, siteName );
-			setData( data );
-			const newSite = data.find( ( site ) => site.path === path );
-			if ( newSite?.id ) {
-				setSelectedSiteId( newSite.id );
+			const tempSiteId = crypto.randomUUID();
+			setData( ( prevData ) =>
+				sortSites( [
+					...prevData,
+					{
+						id: tempSiteId,
+						name: siteName || path,
+						path,
+						running: false,
+						isAddingSite: true,
+						phpVersion: '',
+					},
+				] )
+			);
+			setSelectedSiteId( tempSiteId ); // Set the temporary ID as the selected site
+
+			try {
+				const data = await getIpcApi().createSite( path, siteName );
+				const newSite = data.find( ( site ) => site.path === path );
+				if ( newSite?.id ) {
+					setSelectedSiteId( newSite.id ); // Update the selected site to the new site's ID
+				}
+				setData( ( prevData ) =>
+					sortSites(
+						prevData.map( ( site ) =>
+							site.id === tempSiteId ? { ...site, ...newSite, isAddingSite: false } : site
+						)
+					)
+				);
```

Comment:
Here I suggest one optimization and one fix.
The optimization is to avoid sortingSites again, since it just check the names. And avoid overwriting the ...site, ...newSite and just use the new site.

The bug fix is that if I create a new site, and I select another site in the sidebar, I was switched to the new site automatically.

```suggestion
				if ( ! newSite ) {
					throw new Error( 'Failed to create site' );
				}
				// Update the selected site to the new site's ID if the user didn't change it
				setSelectedSiteId( ( prevSelectedSiteId ) => {
					if ( prevSelectedSiteId === tempSiteId ) {
						return newSite.id;
					}
					return prevSelectedSiteId;
				} );
				setData( ( prevData ) =>
					prevData.map( ( site ) => ( site.id === tempSiteId ? newSite : site ) )
				);
```

If we apply the suggestion, we need to update the `useSelectedSite` hook too:
```
function useSelectedSite( firstSiteId: string | null ) {
	const SELECTED_SITE_ID_KEY = 'selectedSiteId';
	const selectedSiteIdFromLocal = localStorage.getItem( SELECTED_SITE_ID_KEY ) || null;
	const [ selectedSiteId, setSelectedSiteId ] = useState< string | null >(
		selectedSiteIdFromLocal
	);
	useEffect( () => {
		if ( selectedSiteId ) {
			localStorage.setItem( SELECTED_SITE_ID_KEY, selectedSiteId );
		}
	} );

	return {
		selectedSiteId: selectedSiteId || firstSiteId,
		setSelectedSiteId,
	};
}
```

Reply:
Sounds good, I made some changes in https://github.com/Automattic/studio/pull/362/commits/566ef6f2cd5fe2133a8eb97a81701894deff470f and https://github.com/Automattic/studio/pull/362/commits/4bbe32da1993453af41ce5ab18cce25c1ca3e30e to adjust this based on the feedback.